src/MagnumPlugins/AssimpImporter/AssimpImporter.cpp

```diff
@@ -1474,7 +1668,67 @@ Containers::Optional<AnimationData> AssimpImporter::doAnimation(UnsignedInt id)
         Warning{} << "Trade::AssimpImporter::animation(): quaternions in some rotation tracks were renormalized";
 
     return AnimationData{std::move(data), std::move(tracks),
-        mergeAnimationClips ? nullptr : &_f->scene->mAnimations[id]};
+        mergeAnimationClips ? nullptr : _f->scene->mAnimations[id]};
+}
+
+UnsignedInt AssimpImporter::doSkin3DCount() const {
+    /* If the skins are merged, there's at most one */
+    if(_f->mergeSkins)
+        return _f->meshesWithBones.empty() ? 0 : 1;
+
+    return _f->meshesWithBones.size();
+}
+
+Int AssimpImporter::doSkin3DForName(const std::string& name) {
+    /* If the skins are merged, don't report any names */
+    if(_f->mergeSkins) return -1;
+
+    if(!_f->skinsForName) {
+        _f->skinsForName.emplace();
+        _f->skinsForName->reserve(_f->meshesWithBones.size());
+        for(std::size_t i = 0; i != _f->meshesWithBones.size(); ++i)
+            _f->skinsForName->emplace(
+                _f->scene->mMeshes[_f->meshesWithBones[i]]->mName.C_Str(), i);
+    }
+
+    const auto found = _f->skinsForName->find(name);
+    return found == _f->skinsForName->end() ? -1 : found->second;
+}
+
+std::string AssimpImporter::doSkin3DName(const UnsignedInt id) {
+    /* If the skins are merged, don't report any names */
+    if(_f->mergeSkins) return {};
+    return _f->scene->mMeshes[_f->meshesWithBones[id]]->mName.C_Str();
+}
+
+Containers::Optional<SkinData3D> AssimpImporter::doSkin3D(const UnsignedInt id) {
+    /* Import either a single mesh skin or all of them together.
+       Since Assimp gives us no way to enumerate the original skins
+       and assumes that one mesh = one skins, we give the users
```

Comment:
```suggestion
       and assumes that one mesh = one skin, we give the users
```

This is the only remaining issue I found by looking at the code :)